### ISSUE TITLE
Bugs with LBAS attack power formula and AA shot-down

### ISSUE THREAD

**Issue by Divinity123** [noro6/kc-web#8]
Hi, I want to report the following bugs in the air battle simulator.

## 1. LBAS attack power soft cap
According to the newest test on [一式戦 隼III型改(熟練／20戦隊)](https://twitter.com/Divinity_123/status/1651147441607983104), the LBAS attack soft cap is **190**.

## 2. 陸偵補正
With the same test, we found that 陸偵補正 is a post-cap modifier.

## 3. LBAS attack power formula for 深山
The air battle simulator implements the attack power formula for 深山 as `[torpedo power] * sqrt([slot size]) + 25`.
We found several counterexamples of this in TsunDB. 
https://tsunkit.net/api/routing/battles/3132015/replay
https://tsunkit.net/api/routing/battles/660241/replay

Our guess is that the attack power formula is 
```
pre-cap attack power = 0.8 * ([torpedo power] * sqrt(1.8*[slot size]) + 25)
post-cap attack power = [capped attack power] * [contact modifier] * [land-based reconnaissance modifier] * [combined fleet modifier]
```
In other words, 深山 behaves like a regular land-based attack aircraft except for not getting the post-cap 1.8x LBAA modifier.

## 4. Shot-down in 対潜空襲マス
The faraway enemies in 対潜空襲マス (i.e. the Nu-class CVLs) do not contribute to fleet AA and cannot be selected as defenders.
You can verify this yourself in [KCNav](https://tsunkit.net/nav/?version=23042501) by watching replays in 対潜空襲マス (`api_stage2.api_f_lostcount = 0` for all airstrikes and LBAS attacks).

Thank you in advance for taking the time to read this!

**Comment by noro6** [comment]
Thank you for your report.
I will now correct the bug you have pointed out.

I would like to confirm about 陸偵補正.

https://docs.google.com/spreadsheets/d/15ASj7lHpox7GTQRPSFai5aZYTnZN5Wj9Ep3W5Ib8SCg/edit#gid=508712011
According to this result, the formula including 陸偵補正 is
```
post-cap attack power = int([capped attack power] * [land-based reconnaissance modifier]) * [contact modifier] * [combined fleet modifier] * ...
```

This is my guess, but is this formula incorrect?
Your opinion would be appreciated.

**Comment by Divinity123** [comment]
Thank you so much.

@CC_jabberwock has just shown some counterexamples of the formula I posted above.
https://twitter.com/CC_jabberwock/status/1651208177445404672

In order for the formula to fit his data and the data from TsunDB, we have to raise the soft cap to **220** and make 陸偵補正 a **pre-cap** modifier.
I will collect more samples to see if I can reproduce the damage range @CC_jabberwock has observed.
I will update you with the correct formula after that.
Sorry for any confusion caused.

**Comment by Divinity123** [comment]
Hello again. I can confirm that our speculation earlier is correct. The soft cap should be **220** and 陸偵補正 is a **pre-cap** modifier. Evidence for these can be found in the tweet below.
https://twitter.com/Divinity_123/status/1651948848351158273

You can also verify the samples yourself on [kvnav](https://tsunkit.net/nav/6-5?nodes=A&cfComp=XX&start=2023-04-15) (click on the `Battles` tab in node A). 

Thanks in advance for fixing the bugs!

**Comment by noro6** [comment]
Bug fixes have been completed.
Many thanks for pointing out the bug and for the additional testing!
